pkg/core/resources/model/resource.go

```diff
@@ -435,6 +436,66 @@ func ResourceOrigin(rm ResourceMeta) (mesh_proto.ResourceOrigin, bool) {
 	return "", false
 }
 
+func ComputeLabels(r Resource, mode config_core.CpMode, isK8s bool, systemNamespace string) map[string]string {
+	labels := r.GetMeta().GetLabels()
+	if len(labels) == 0 {
+		labels = map[string]string{}
+	}
+
+	setIfNotExist := func(k, v string) {
+		if _, ok := labels[k]; !ok {
+			labels[k] = v
+		}
+	}
+
+	if isK8s && r.Descriptor().Scope == ScopeMesh {
+		setIfNotExist(metadata.KumaMeshLabel, DefaultMesh)
+	}
+
+	if mode == config_core.Zone {
+		setIfNotExist(mesh_proto.ResourceOriginLabel, string(mesh_proto.ZoneResourceOrigin))
+	}
+
+	if ns, ok := labels[mesh_proto.KubeNamespaceTag]; ok && r.Descriptor().IsPolicy && r.Descriptor().IsPluginOriginated {
+		var role mesh_proto.PolicyRole
+		switch ns {
+		case systemNamespace:
+			role = mesh_proto.SystemPolicyRole
+		default:
+			role = ComputePolicyRole(r.GetSpec().(Policy))
+		}
+		setIfNotExist(mesh_proto.PolicyRoleLabel, string(role))
+	}
+
+	return labels
+}
+
+func ComputePolicyRole(p Policy) mesh_proto.PolicyRole {
+	hasTo := false
+	if pwtl, ok := p.(PolicyWithToList); ok && len(pwtl.GetToList()) > 0 {
+		hasTo = true
+	}
+
+	hasFrom := false
+	if pwfl, ok := p.(PolicyWithFromList); ok && len(pwfl.GetFromList()) > 0 {
+		hasFrom = true
+	}
+
+	if hasTo && !hasFrom {
+		// todo(lobkovilya): detect if the policy is a producer policy when they're supported
+		return mesh_proto.ConsumerPolicyRole
+	} else {
+		return mesh_proto.WorkloadOwnerPolicyRole
+	}
+}
```

Comment:
Do we need to define a `default` return? 
It seems like `if hasTo == false && hasFrom == false`, we'll return an empty string "" ?

Reply:
why? If I read it correctly, we will return either `mesh_proto.ConsumerPolicyRole` or `mesh_proto.WorkloadOwnerPolicyRole` there are no other return statements in this function

Reply:
Oh, you're right. 
My fault, my head was foggy.